### PR TITLE
Setup Custom metrics

### PR DESCRIPTION
**Pull request by andyduong1920**
[Set up custom metric for LiveDashboard](https://www.notion.so/nimblehq/Set-up-custom-metric-for-LiveDashboard-ed5fafc7d2e649b0ba06783b6eb8e4fb)

## What happened

Add custom metric on the submit answer event.
 
## Insight

We would like to track the submit duration (from the challenge started moment), with two metadata (username and challenge name).

> For the challenger start time, we didn't have it yet so we can set it at `24/11/2021 - 15:40:00` for the demo purpose, later on after this Nimble Growth, if we would like to have that feature, we could store the strated_time in the `challenges` table.

Emit 2 events:
- submission -> Total submission
- valid_submission -> Valid submission

From these events, we will let the LiveDashboard display these metric by define these listeners

image

 
## Proof Of Work

[Videocast](https://drive.google.com/file/d/1hdtZyE621TP0e6WzQ-gzHMFDKtp8_cnG/view?usp=sharing)

1/ Counter total submission - per question

image

2/ Counter total submission - per user_name across all questions

image

3/ Counter total valid submission - per question

image

4/ Summary the valid submission duration per each question -> We can see the average submission time per each question

image

### CODE REVIEW

File: lib/codewar_web/live/home/index_live.ex [hunk 1]
```diff
@@ -33,8 +34,23 @@ defmodule CodewarWeb.Home.IndexLive do
       ) do
     with %Ecto.Changeset{changes: answer_params, valid?: true} = changeset <-
            AnswerSubmissionForm.new(params),
-         {:ok, answer} <- Competitions.create_answer(answer_params) do
+         {:ok, %{challenge: %{name: challenge_name}} = answer} <-
+           Competitions.create_answer(answer_params) do
+      # On 24/11/2021 - 15:40:00
+      challenger_started_time = 1_637_743_147_420_001_000
```
Comment (rosle) [inline lib/codewar_web/live/home/index_live.ex:40]: Just wondering why we not use `started_at` column in challenge 🤔

File: lib/codewar_web/live/home/index_live.ex [hunk 2]
```diff
@@ -33,8 +34,23 @@ defmodule CodewarWeb.Home.IndexLive do
       ) do
     with %Ecto.Changeset{changes: answer_params, valid?: true} = changeset <-
            AnswerSubmissionForm.new(params),
-         {:ok, answer} <- Competitions.create_answer(answer_params) do
+         {:ok, %{challenge: %{name: challenge_name}} = answer} <-
```
Comment (rosle) [inline lib/codewar_web/live/home/index_live.ex:37]: Without preload, I think we can simply do `get_challenge(params.challenge_id)` here to get challenge info.

**Review by rosle — APPROVED** [review]
Just couple of questions/suggestions. But I'm ok if you wanna just merge it for the sake of the demo 😆 👍

File: lib/codewar_web/live/home/index_live.ex (hunk 2 above)
Comment (andyduong1920) [reply lib/codewar_web/live/home/index_live.ex:37]: For the demo purpose will keep it as preload for now @rosle

File: lib/codewar_web/live/home/index_live.ex (hunk 1 above)
Comment (andyduong1920) [reply lib/codewar_web/live/home/index_live.ex:40]: Oh, I don't know that field honestly :D let me check it

Comment (andyduong1920) [reply lib/codewar_web/live/home/index_live.ex]: Used the started_at field in https://github.com/nimblehq/codewar-web/pull/83/commits/5476377172a279c0124d47554095c8745ba4bffa, and tested it on my local @rosle 



image

**Pull request merged**